packages/block-library/src/query-pagination/edit.js

```diff
@@ -53,9 +54,11 @@ export default function QueryPaginationEdit( {
 		allowedBlocks: ALLOWED_BLOCKS,
 	} );
 	// Always show label text if paginationArrow is set to 'none'.
-	if ( paginationArrow === 'none' ) {
-		setAttributes( { showLabel: true } );
-	}
+	useEffect( () => {
+		if ( paginationArrow === 'none' ) {
+			setAttributes( { showLabel: true } );
+		}
+	}, [ paginationArrow, setAttributes ] );
```

Comment:
We need `showLabel` in deps.

Reply:
Got it - done in https://github.com/WordPress/gutenberg/pull/51165/commits/e7a9fa5089cc86717317131a03795f5cc63f07e0.